active-rfcs/0000-vtu-api.md

```diff
@@ -0,0 +1,268 @@
+- Start Date: 19.04.2020
+- Target Major Version: 2.x
+- Reference Issues: (fill in existing related issues, if any)
+- Implementation PR: (leave this empty)
+
+# Summary
+
+VueTestUtils 2.x, which targets Vue 3, will introduce a few new methods and remove some less used ones.
+
+- **Breaking:** `sync` mode removed. All `wrapper` methods return `nextTick`, you must `await` them
```

Comment:
I believe `must` is too restrictive here - there *might* be use cases when you wasn't to perform multiple actions. Maybe we could replace this one with `can`.

Also `await` is also a bit "opinionated" - not all codebase are using `async / await`. Mentioning that it returns Promise might be better

Reply:
Correct! Users can choose to not await these, the dom would not be yet updated. Will reword it :)

Reply:
Being opinionated here is totally fine.
I think recommending people to `await` those is better than finding yourself not knowing why your test is passing (or not).
The sentence also makes sense given it's noted alongside the removal of `sync` mode, which makes this _awaiting_ necessary.

Recommending `await` against `then` is pretty common in test environments:

```js
it('test', async () => {
  await wrapper.setProps({ one: 'one' })
  expect(wrapper.find('a').classes()).toBe('active')
  await wrapper.setProps({ two: 'two' })
  expect(wrapper.find('a').classes()).not.toBe('active')
})
```

vs

```js
it('test', () => {
  return wrapper.setProps({ one: 'one' }).then(() => {
    expect(wrapper.find('a').classes()).toBe('active')
    return wrapper.setProps({ two: 'two' })
  }).then(() => {
    expect(wrapper.find('a').classes()).not.toBe('active')
  })
})
```

To make the sentence clear, it could say 

> return a promise returned by `nextTick()`

or something similar

Reply:
@posva Hmm, while I definitely support `async / await` as being way more readable, I was confused by assumption, that since Vue3 is targeting (also) IE11, VTU will follow and will not use any syntax that target platform does not support

If this is intended decision to use `async / await` elsewhere it's ok :)